src/rgw/rgw_kms.cc

```diff
@@ -312,7 +313,7 @@ static int request_key_from_barbican(CephContext *cct,
     return -EINVAL;
   }
   concat_url(secret_url, "/v1/secrets/");
-  concat_url(secret_url, std::string(key_id));
+  concat_url(secret_url, std::string(url_encode(key_id)));
```

Comment:
url_encode() returns std::string, so you don't need to duplicate that here

Reply:
@matthewoliver could you address this comment?

Reply:
No idea why am still type casting to the std::string. It's been a while since I wrote the code.. and would've thought I tested and did it for a reason. But that reason has long gone.

The type case was to change the string_view to the string for concat_url. And @cbodley is right, url_encode returns a std::string. So have rebased this PR with master and removed the type cast.

Maybe I was just too curious back when I did this patch as I was new to the codebase ???